cocos/base/CCDirector.cpp

```diff
@@ -305,6 +309,13 @@ void Director::drawScene()
             physics3DWorld->stepSimulate(_deltaTime);
         }
 #endif
+#if CC_USE_NAVMESH
+        auto navMesh = _runningScene->getNavMesh();
+        if (navMesh)
+        {
+            navMesh->update(_deltaTime);
+        }
```

Comment:
I would decouple `navMesh` from `Director`.
In anycase, `Scene` should call `_navMesh->update()` in their own `update()` method. That means that `Scene` should schedule its own update method when a `navMesh` is added into it.